demo-apps/cf-secure/src/main/java/org/eclipse/californium/examples/CredentialsUtil.java

```diff
@@ -0,0 +1,180 @@
+/*******************************************************************************
+ * Copyright (c) 2017 Bosch Software Innovations GmbH and others.
+ * 
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * and Eclipse Distribution License v1.0 which accompany this distribution.
+ * 
+ * The Eclipse Public License is available at
+ *    http://www.eclipse.org/legal/epl-v10.html
+ * and the Eclipse Distribution License is available at
+ *    http://www.eclipse.org/org/documents/edl-v10.html.
+ * 
+ * Contributors:
+ *    Bosch Software Innovations GmbH - initial implementation
+ ******************************************************************************/
+
+package org.eclipse.californium.examples;
+
+import java.io.IOException;
+import java.security.GeneralSecurityException;
+import java.security.cert.Certificate;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.List;
+
+import org.eclipse.californium.elements.util.SslContextUtil;
+import org.eclipse.californium.scandium.config.DtlsConnectorConfig;
+import org.eclipse.californium.scandium.dtls.pskstore.InMemoryPskStore;
+
+/**
+ * Credentials utility for setup DTLS credentials.
+ */
+public class CredentialsUtil {
+
+	/**
+	 * Credentials mode.
+	 */
+	public enum Mode {
+		/**
+		 * Preshared secret keys
+		 */
+		PSK,
+		/**
+		 * Raw public key certificates
+		 */
+		RPK,
+		/**
+		 * X.509 certificates
+		 */
+		X509
+	}
+
+	/**
+	 * Default list of modes.
+	 * 
+	 * Value is PSK, RPK, X509.
+	 */
+	public static final List<Mode> DEFAULT_MODES = Arrays.asList(new Mode[] { Mode.PSK, Mode.RPK, Mode.X509 });
+
+	// from ETSI Plugtest test spec
+	public static final String PSK_IDENTITY = "password";
+	public static final byte[] PSK_SECRET = "sesame".getBytes();
+
+	// from demo-certs
+	public static final String SERVER_NAME = "server";
+	public static final String CLIENT_NAME = "client";
+	private static final String TRUST_NAME = "root";
+	private static final char[] TRUST_STORE_PASSWORD = "rootPass".toCharArray();
+	private static final char[] KEY_STORE_PASSWORD = "endPass".toCharArray();
+	private static final String KEY_STORE_LOCATION = "certs/keyStore.jks";
+	private static final String TRUST_STORE_LOCATION = "certs/trustStore.jks";
+
+	/**
+	 * Parse arguments to modes.
+	 * 
+	 * @param args arguments
+	 * @return array of modes. if the arguments are empty,
+	 *         {@link #DEFAULT_MODES} is returned.
+	 */
+	public static List<Mode> parse(String[] args) {
+		int len = args.length;
+		if (0 == len) {
```

Comment:
Still remains a MISRA's memory :) (this is one is less confusing)

I think the use of  `len` variable is not really needed. 
Unless you eventually reuse it below in `new ArrayList<>(len)`, personnaly I prefer to use args.length in the first `if`